Swiftfin/Views/ItemEditorView/IdentifyItemView/IdentifyItemView.swift

```diff
@@ -0,0 +1,340 @@
+//
+// Swiftfin is subject to the terms of the Mozilla Public
+// License, v2.0. If a copy of the MPL was not distributed with this
+// file, you can obtain one at https://mozilla.org/MPL/2.0/.
+//
+// Copyright (c) 2024 Jellyfin & Jellyfin Contributors
+//
+
+import Combine
+import JellyfinAPI
+import SwiftUI
+
+struct IdentifyItemView<SearchInfo: Equatable>: View {
+
+    // MARK: - Observed & Environment Objects
+
+    @EnvironmentObject
+    private var router: ItemEditorCoordinator.Router
+
+    @ObservedObject
+    private var viewModel: ItemInfoViewModel<SearchInfo>
+
+    // MARK: - Identity Variables
+
+    @State
+    private var lastSearch: SearchInfo?
+    @State
+    private var tempSearchInfo: SearchInfo?
+    @State
+    private var selectedMatch: RemoteSearchResult?
+
+    // MARK: - Error State
+
+    @State
+    private var error: Error?
+
+    // MARK: - Potential States
+
+    @State
+    var boxSetInfo = BoxSetInfo()
```

Comment:
As much as its annoying that there are these objects that are the exact same structure for each item type, we don't need to have variables for each one.

The fields themselves should have their own source and then the view models receive the strings and construct their own info objects.

Reply:
Sounds good! That doesn't seem too bad. I was having trouble wrapping my head around all this. Is that right from the coordinator or should I put the switch on the View instead?

Reply:
Outside of making the objects, there shouldn't really be any `switch`ing in `IdentifyItemView` as the objects are subclassed. If the base search method instead took as parameters the fields being filled out then they would instead construct the necessary `_Info` objects for their eventual request.

Reply:
I took a swing at this. Since I have the `BaseItemDto` in the `ViewModel`, I changed this to take the generic title, original title, & year then switch inside of the `ViewModel` based on the type. This lets me use the view and the viewModel without specifying type.

I *think* this is better? Less code and it works as expected. Open to feedback if you think there is a better route!